Framework/Algorithms/src/CreateWorkspace.cpp

```diff
@@ -106,9 +106,12 @@ void CreateWorkspace::exec() {
   const std::string vUnit = getProperty("VerticalAxisUnit");
   const std::vector<std::string> vAxis = getProperty("VerticalAxisValues");
 
-  if ((vUnit != "SpectraNumber") && (static_cast<int>(vAxis.size()) != nSpec)) {
-    throw std::invalid_argument(
-        "Number of y-axis labels must match number of histograms.");
+  const int vAxisSize = static_cast<int>(vAxis.size());
+  if (vUnit != "SpectraNumber" &&
```

Comment:
A comment here explaining what it does would be nice, from a brief look it's very hard to interpret what the many nested conditions do. Maybe a refactoring of the condition would be better